tests/queries/tests.py

```diff
@@ -2016,6 +2016,17 @@ def test_ordered_subselect(self):
         query = DumbCategory.objects.filter(id__in=DumbCategory.objects.order_by('-id')[2:])
         self.assertEqual(set(query.values_list('id', flat=True)), {1, 2})
 
+    def test_ordered_select_for_update(self):
+        """
+        Subqueries containing select_for_update should respect ordering. Users
+        may use the order_by clause to specify rows locking order and prevent
+        deadlocks.
+
+        See https://code.djangoproject.com/ticket/27193 for more information.
+        """
+        qs = DumbCategory.objects.filter(id__in=DumbCategory.objects.order_by('-id').select_for_update())
```

Comment:
It would be better to put this test in `tests/select_for_update` -- queries is more for things that don't fit anywhere more specific.